src/Microsoft.Health.Dicom.Core/Features/Validation/Errors/ElementValidationError.cs

```diff
@@ -0,0 +1,46 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Dicom;
+using EnsureThat;
+
+namespace Microsoft.Health.Dicom.Core.Features.Validation.Errors
+{
+    public abstract class ElementValidationError
```

Comment:
Should all of these "error" classes simply be derived `Exceptions` from `DicomElementValidationException`? In other words, should we merge `DicomElementValidationException` and `ElementValidationError`?

Reply:
I'm trying to encapsulate error information into a separated class, so that in the future, if we want both Store and Reindex have same error message, can easily jsonify into string, and store into some store( like SQL or table storage).

From another point, this might be overkill for now.